pywbemtools/pywbemcli/pywbemcli.py

```diff
@@ -255,14 +255,87 @@ def _validate_no_server_option_conficts(server, connection_name, mock_server):
-                            user, password, resolved_timeout,
-                            resolved_use_pull, resolved_pull_max_cnt,
+def _get_default_connection(connections_repo, verbose):
+    """
+    Attempt to get the default connection name from the connections repo
+    if there is a repo defined. if there is no repo or no default connection
+    name in the repo, returns None.
+
+    Returns:
+        String containing valid connection name that is the default connection
+        or None if there is no connections repo or no default connection
+        defined in the repo
+
+    Raises:
+        click.ClickException if the default name exists but the corresponding
+        connection definition does not exist.
+    """
+    try:
+        if connections_repo.file_exists():
+            connection_name = connections_repo.default_connection_name
+
+            # Test if the default connection name is actually
+            # in the repo.  If not reset the default connection
+            # name to None and continue.
+            # This should never occur unless connection file
+            # is corrupted.
+            if connection_name and connection_name not in \
+                    connections_repo:
+                connections_repo.default_connection_name = None
+                raise click.ClickException(
+                    'Default connection name: "{}" not found in '
+                    'connections file "{}"; default connection '
+                    'name cleared.'.
+                    format(connection_name,
+                           connections_repo.connections_file))
+            if verbose:
+                click.echo('Current connection: "{}"'
+                           .format(connection_name))
+            return connection_name
+
+        # returns no if there is no file
+        return None
+
+    # Account for other exception from file access
+    except ConnectionsFileError as cfc:
+        click.echo("{}, {}". format(cfc.__class__.__name__, cfc),
+                   err=True)
+        raise click.Abort()
+
+
+def _validate_server_only_options(pywbem_server, user, password, timeout,
+                                  verify, certfile, keyfile, ca_certs):
+    """
+    Validate that the options defined as parameters for this function
+    are not set if the server type is mock_server
+    """
+    if pywbem_server.mock_server:
+        opts = [user, password, timeout, verify is not None, certfile,
+                keyfile, ca_certs]
+        if any(opts):
+            opt_names = ["--user", "--password", "--timeout",
+                         "--verify", "--certfile", "--keyfile",
+                         "--ca_certs"]
+            errors = [opt_names[c] for c, i in enumerate(opts) if i]
+            raise click.ClickException(
+                "General option(s): ({0}) not allowed with --mock-server "
+                "general option'".
```

Comment:
First, superflous trailing single quote in exception message.

Second, in case of specifying a named connection, the mock_server attribute of PywbemServer is set from the connections repo, but it was not specified in the command line, and thus the error message is misleading.

You can try that out with:
```
$ pywbemcli -n simple_interop -u myuser
Error: General option(s): (--user) not allowed with --mock-server general option'
```

To address that, I suggest changing "not allowed with --mock-server general option'" to "not allowed when using a mock server".

Reply:
TKS. Completely missed that one.   DONE